operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/internal/CustomResourceEventSource.java

```diff
@@ -135,28 +117,25 @@ public void eventReceived(Watcher.Action action, CustomResource customResource)
     }
   }
 
-  private void markLastGenerationProcessed(CustomResource resource) {
+  private void markLastGenerationProcessed(T resource) {
     if (generationAware && resource.hasFinalizer(resourceFinalizer)) {
       lastGenerationProcessedSuccessfully.put(
           KubernetesResourceUtils.getUID(resource), resource.getMetadata().getGeneration());
     }
   }
 
-  private boolean skipBecauseOfGenerations(CustomResource customResource) {
+  private boolean skipBecauseOfGeneration(T customResource) {
     if (!generationAware) {
       return false;
     }
     // if CR being deleted generation is naturally not changing, so we process all the events
     if (customResource.isMarkedForDeletion()) {
       return false;
     }
-    if (!largerGenerationThenProcessedBefore(customResource)) {
-      return true;
-    }
-    return false;
+    return !hasGenerationAlreadyBeenProcessed(customResource);
   }
 
-  public boolean largerGenerationThenProcessedBefore(CustomResource resource) {
+  private boolean hasGenerationAlreadyBeenProcessed(T resource) {
```

Comment:
Actually the I would keep the old one because:
1. that is what we actually doing (comparing if its larger)
2. In general this should be linear increasing number because of the strong consistency guarantees of etcd. Thus we can expect that that we receive these events in order. But checking for a large number makes it a nice defensive programming practice.

Reply:
this is quite low level code so it's not bad it exposes the intention. And hasGenerationAlreadyBeenProcessed is clearly misleading, it can happen that the generation is not processed before, like when operator was down.

Reply:
this is quite low level code so it's not bad it exposes the intention. And hasGenerationAlreadyBeenProcessed is clearly misleading, it can happen that the generation is not processed before, like when operator was down.

Reply:
Yes but the intent of the method is still to figure out whether the generation of the provided `CustomResource` has been seen/processed already.  
It also doesn't matter if the method if low level or not. People still need to read the code and figure out what its intent is. An alternative would be to remove that method altogether and merge it in the calling code (since it's only called in one spot).

Reply:
I don't like these long methods, but I think the naming pretty much should reflect what is happening in the implementation, if its higher level abstraction is better, but should not be confusing. If sombody reads this, and checks the implementation, could think if the generation is higher all the previouse generations where processed, what is cleary not the case.

note that this method already just called from a higher abstraction the "skipBecauseOfGeneration". So when I read it:
ok we are skipping when this generation is not larger then we already seen before. So its reads quite well as it is now.

Reply:
But clearly the implementation has nothing to do with processing or not, its about just the generation increased.

Reply:
So what is the conclusion here?
Still think this new name is misleading.

Reply:
Inlined the method :)